esphome/components/canbus/__init__.py

```diff
@@ -132,6 +134,11 @@ async def canbus_action_to_code(config, action_id, template_arg, args):
     )
     cg.add(var.set_use_extended_id(use_extended_id))
 
+    remote_transmission_request = await cg.templatable(
+        config[CONF_REMOTE_TRANSMISSION_REQUEST], args, cg.uint32
```

Comment:
Why is this set to `cg.uint32`?
```suggestion
        config[CONF_REMOTE_TRANSMISSION_REQUEST], args, bool
```